charmcraft/commands/store/charmlibs.py

```diff
@@ -29,10 +28,20 @@
 
 from charmcraft.errors import BadLibraryNameError, BadLibraryPathError
 
-LibData = namedtuple(
-    "LibData",
-    "lib_id api patch content content_hash full_name path lib_name charm_name",
-)
+
+@dataclass(frozen=True)
+class LibData:
```

Comment:
do you think there's any chance that this change breaks someone somehow?

Reply:
I shouldn't think so. This file contains helpers for the various `*-lib` store commands, so it would have to be someone who's using our internal data structures and trying to iterate over a LibData object (which one shouldn't be doing anyway). The fields are kept in the same order so the constructors are the same.

If we do desire to keep it a NamedTuple still I could convert it into a typed NamedTuple, but I would be surprised if there's any direct external use of this class.